ITISFoundation/osparc-simcore
# ♻️ update settings api server

## What do these changes do?

- ♻️ refactors ``api-server`` to use ``settings-library`` (based on pattern presented in http://github.com/pcrespov/sample-app-settings)
- ♻️ formatted ``meta`` constants to uppercase in both ``api-server`` and ``director-v2``
- ✨adds CLI command to resolve and print envfiles
- 🐛added a retry policy to retrieve index from static web server since i noticed that  unhandled exception shut down the web-server in one of the tests:
```
   aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host static-webserver:8000 ssl:default [Name or service not known]
```
## Related issue/s

- Follows from PR #2558
- 

## How to test

- all tests in api-server
- check CLI to produce e.g. *envfiles*  
```
$ cd services/api-server
$ make install-dev
$ simcore-service-api-server --help   
...
$ simcore-service-api-server settings --help
...
$ export $(grep -v '^#' .env-devel | xargs),  simcore-service-api-server settings --compact
```

## Checklist

check and update settings in
- [x] ``core/settings.py``
- [x] ``.env-dev``
- [x] ``docker-compose``
